mobile/lib/shared/providers/haptic_feedback.provider.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/services.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/modules/settings/providers/app_settings.provider.dart';
+import 'package:immich_mobile/modules/settings/services/app_settings.service.dart';
+
+final hapticFeedbackProvider =
+    StateNotifierProvider<HapticNotifier, void>((ref) {
+  return HapticNotifier(ref);
+});
+
+class HapticNotifier extends StateNotifier<void> {
+  void build() {}
+  final Ref _ref;
+
+  HapticNotifier(this._ref) : super(null);
+
+  selectionClick() {
+    if (_ref
+        .read(appSettingsServiceProvider)
+        .getSetting(AppSettingsEnum.enableHapticFeedback)) {
```

Comment:
We can move all such settings read directly to the state, so instead of void, the state would be bool and then all the method calls will check for truthfulness of the state instead of fetching them each time

Reply:
Good idea!

Reply:
Can you elaborate on how would you do that, I tried a few technique but it doesn't seem to register the changes when the setting changes